library/src/animationManager.cxx

```diff
@@ -282,11 +286,70 @@ bool animationManager::LoadAtTime(double timeValue)
   }
   return true;
 }
+void animationManager::CycleAnimation()
+{
+  assert(this->Importer);
+  if (this->AvailAnimations <= 0)
+  {
+    return;
+  }
+
+  this->DisableAllAnimation();
 
+  if (this->AnimationIndex == this->AvailAnimations - 1)
+  {
+    this->AnimationIndex = -1;
+  }
+  else
+  {
+    this->AnimationIndex += 1;
+  }
+  this->EnableAllAnimation();
+  
+  this->LoadAtTime(this->TimeRange[0]);
+}
+// ---------------------------------------------------------------------------------
+int animationManager::GetAnimationIndex()
+{
+  return this->AnimationIndex;
+}
+// ---------------------------------------------------------------------------------
+std::string animationManager::GetAnimationName()
+{
+  if (this->AnimationIndex == -1)
+  {
+    return "All Animations";
+  }
+  return this->Importer->GetAnimationName(this->AnimationIndex);
+}
 //----------------------------------------------------------------------------
-void animationManager::GetTimeRange(double timeRange[2])
+void animationManager::EnableAllAnimation()
 {
-  timeRange[0] = this->TimeRange[0];
-  timeRange[1] = this->TimeRange[1];
```

Comment:
@kathleenhang you shoukd not remote the GetTimeRange method, It was added in another PR.

Reply:
Fixed! Accidentally removed it during cherrypick merge conflicts since I am new to git.